main.lua

```diff
@@ -1366,3 +1350,71 @@ function userdataIterator(data)
         if success then return idx, item end
     end
 end
+
+local LSPCmds = {
+    ["start"] = startServer,
+    ["stop"] = stopServers,
+    ["showlog"] = showLog,
+    ["sync-document"] = function (bp) syncFullDocument(bp.Buf) end,
+    ["hover"] = hoverAction,
+    ["format"] = formatAction,
+    ["autocomplete"] = completionAction,
+    ["goto-definition"] = gotoAction("definition"),
+    ["goto-declaration"] = gotoAction("declaration"),
+    ["goto-typedefinition"] = gotoAction("typeDefinition"),
+    ["goto-implementation"] = gotoAction("implementation"),
+    ["find-references"] = findReferencesAction,
+    ["document-symbols"] = documentSymbolsAction,
+    ["diagnostic-info"] = openDiagnosticBufferAction,
+}
+
+local function lspCompleter(buf)
+    local opts = {}
+
+    --Do NOT autocomplete after first argument
+    local args = go_strings.Split(buf:Line(0), " ")
+    if #args > 2 then return nil, nil end
+
+    for k, _ in pairs(LSPCmds) do table.insert(opts, k) end
+    table.sort(opts)
```

Comment:
Creating & sorting this new table every time feels a bit silly. I'd rather hard-code a pre-sorted table of completions. That way we don't necessarily need to follow lexicographical order, and it would make it easier to customize the completions depending on the plugin state. For example there is no need to autocomplete "lsp stop" if there are no language servers running.

Reply:
It is, but some people prefer not to hardcode values, and I wasn't aware of your preference. I will add a note indicating that it is hardcoded.